menas/src/main/scala/za/co/absa/enceladus/menas/controllers/VersionedModelController.scala

```diff
@@ -159,15 +161,19 @@ abstract class VersionedModelController[C <: VersionedModel with Product with Au
   }
 
   @PutMapping(Array("/lock/{name}"))
+  @PreAuthorize("@authConstants.hasAdminRole(authentication)")
   @ResponseStatus(HttpStatus.OK)
-  def lock(@PathVariable name: String): CompletableFuture[UpdateResult] = {
-    versionedModelService.setLock(name, isLocked = true)
+  def lock(@PathVariable name: String,
```

Comment:
In all other API cases (except one) the @AuthenticationPrincipal is the first param. I would keep it consistent.